pkg/queue/manager.go

```diff
@@ -48,14 +51,18 @@ type Manager struct {
 	clusterQueues map[string]ClusterQueue
 	localQueues   map[string]*LocalQueue
 
+	workloadsStatus *kueue.ClusterQueuePendingWorkloadsStatus
+	cfg             *configapi.Configuration
+
 	// Key is cohort's name. Value is a set of associated ClusterQueue names.
 	cohorts map[string]sets.Set[string]
 }
 
-func NewManager(client client.Client, checker StatusChecker) *Manager {
+func NewManager(client client.Client, checker StatusChecker, cfg *configapi.Configuration) *Manager {
```

Comment:
Pass individual settings, instead of the entire config.

You can follow the options pattern, like we do for the reconciler:

https://github.com/kubernetes-sigs/kueue/blob/6d30b82d1459590cdc553d9eb8f88f553635500b/pkg/controller/jobframework/reconciler.go#L68

Reply:
added options